kotlinx-coroutines-core/common/src/internal/DispatchedTask.kt

```diff
@@ -205,3 +203,14 @@ internal inline fun DispatchedTask<*>.runUnconfinedEventLoop(
 internal inline fun Continuation<*>.resumeWithStackTrace(exception: Throwable) {
     resumeWith(Result.failure(recoverStackTrace(exception, this)))
 }
+
+/**
+ * This exception holds an exception raised in [CoroutineDispatcher.dispatch] method
```

Comment:
Please leave a comment that helps readers/maintainers to understand **why** it is here without looking into git blame and/or corresponding tests.
For example:

```suggestion
 * This exception holds an exception raised in [CoroutineDispatcher.dispatch] method.
 * When dispatcher methods fail unexpectedly, it is likely a user-induced programmatic bug,
 * such as calling `executor.close()` prematurely. To avoid reporting such exceptions as fatal errors,
 * we handle them with a separate code path. See also #4091.
```